client/my-sites/customer-home/cards/launchpad/keep-building.tsx

```diff
@@ -29,94 +28,27 @@ const LaunchpadKeepBuilding = ( { site }: LaunchpadKeepBuildingProps ): JSX.Elem
 	const numberOfSteps = checklist?.length || 0;
 	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
 	const tasklistCompleted = completedSteps === numberOfSteps;
-
-	const recordTaskClickTracksEvent = ( task: Task ) => {
-		recordTracksEvent( 'calypso_launchpad_task_clicked', {
-			checklist_slug: checklistSlug,
-			checklist_completed: tasklistCompleted,
-			task_id: task.id,
-			is_completed: task.completed,
-			context: 'customer-home',
-		} );
-	};
+	const tracksData = { recordTracksEvent, checklistSlug, tasklistCompleted };
 
 	const [ shareSiteModalIsOpen, setShareSiteModalIsOpen ] = useState( false );
 
 	const sortedTasksWithActions = ( tasks: Task[] ) => {
-		const completedTasks = tasks.filter( ( task: Task ) => task.completed );
-		const incompleteTasks = tasks.filter( ( task: Task ) => ! task.completed );
+		const tasksWithActions = setUpActionsForTasks( tasks, siteSlug, tracksData );
+
+		// Add action to `share_site` task, which has a custom UI.
+		const shareSiteTask = tasksWithActions.find( ( task: Task ) => task.id === 'share_site' );
+		if ( shareSiteTask ) {
+			shareSiteTask.actionDispatch = () => {
+				setShareSiteModalIsOpen( true );
+			};
+		}
+
+		const completedTasks = tasksWithActions.filter( ( task: Task ) => task.completed );
+		const incompleteTasks = tasksWithActions.filter( ( task: Task ) => ! task.completed );
 
 		const sortedTasks = [ ...completedTasks, ...incompleteTasks ];
```

Comment:
We could simply return without using the `const`.
```suggestion
		return [ ...completedTasks, ...incompleteTasks ];
```